include/clang/AST/Expr.h

```diff
@@ -4748,27 +4752,68 @@ class CountBoundsExpr : public BoundsExpr {
   }
 };
 
+class RelativeBoundsExpr : public BoundsExpr {
```

Comment:
Could you rename to this class to RelativeBoundsClause?

I don't think  it needs to be subclass of BoundsExpr or an expression at all.  It's a helper class and doesn't make much sense as an expression.

It would be helpful to add a comment describing why we have it.  There will be two different ways of specifying relative alignment, so we have a class to encapsulate it.  In addition, we expect bounds declarations with relative alignment clauses to be rare, so having the data out-of-line reduces memory usage.